pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,511 @@
+	committedRec := 0
+	for {
+		select {
+		case <-ch.stopChan:
+			commitTicker.Stop()
+			logTicker.Stop()
+			return
+		case <-commitTicker.C:
+			committed, err := ch.batchCommitAll()
+			if err == nil {
+				committedRec += committed
+			}
+		case <-logTicker.C:
+			klog.V(4).InfoS("Total number of records committed to DB", "count", committedRec)
+			committedRec = 0
+		}
+	}
+}
+
+func (ch *ClickHouseExportProcess) batchCommitAll() (int, error) {
+	currSize := ch.deque.Len()
+	if currSize == 0 {
+		return 0, nil
+	}
+
+	var stmt *sql.Stmt
+
+	// start new connection
+	tx, err := ch.db.Begin()
+	if err == nil {
+		stmt, err = tx.Prepare(insertQuery)
+	}
+	if err != nil {
+		klog.Errorf("error when preparing insert statement: %v", err)
+		_ = tx.Rollback()
+		return 0, err
+	}
+
+	// populate items from deque
+	validCount := 0
+	for i := 0; i < currSize; i++ {
+		record, ok := ch.deque.At(i).(ClickHouseFlowRow)
+		if !ok {
+			continue
+		}
+		_, err := stmt.Exec(
+			record.flowStartSeconds,
+			record.flowEndSeconds,
+			record.flowEndSecondsFromSourceNode,
+			record.flowEndSecondsFromDestinationNode,
+			record.flowEndReason,
+			record.sourceIP,
+			record.destinationIP,
+			record.sourceTransportPort,
+			record.destinationTransportPort,
+			record.protocolIdentifier,
+			record.packetTotalCount,
+			record.octetTotalCount,
+			record.packetDeltaCount,
+			record.octetDeltaCount,
+			record.reversePacketTotalCount,
+			record.reverseOctetTotalCount,
+			record.reversePacketDeltaCount,
+			record.reverseOctetDeltaCount,
+			record.sourcePodName,
+			record.sourcePodNamespace,
+			record.sourceNodeName,
+			record.destinationPodName,
+			record.destinationPodNamespace,
+			record.destinationNodeName,
+			record.destinationClusterIP,
+			record.destinationServicePort,
+			record.destinationServicePortName,
+			record.ingressNetworkPolicyName,
+			record.ingressNetworkPolicyNamespace,
+			record.ingressNetworkPolicyRuleName,
+			record.ingressNetworkPolicyRuleAction,
+			record.ingressNetworkPolicyType,
+			record.egressNetworkPolicyName,
+			record.egressNetworkPolicyNamespace,
+			record.egressNetworkPolicyRuleName,
+			record.egressNetworkPolicyRuleAction,
+			record.egressNetworkPolicyType,
+			record.tcpState,
+			record.flowType,
+			record.sourcePodLabels,
+			record.destinationPodLabels,
+			record.throughput,
+			record.reverseThroughput,
+			record.throughputFromSourceNode,
+			record.throughputFromDestinationNode,
+			record.reverseThroughputFromSourceNode,
+			record.reverseThroughputFromDestinationNode)
+
+		if err != nil {
+			klog.Errorf("error when adding record: %v", err)
+			_ = tx.Rollback()
+			return 0, err
+		}
+		validCount += 1
+	}
+
+	if err := tx.Commit(); err != nil {
+		klog.Errorf("error when committing record: %v", err)
+		return 0, err
+	}
+
+	// remove committed record from deque
+	ch.mutex.Lock()
+	defer ch.mutex.Unlock()
+	for i := 0; i < currSize; i++ {
+		ch.deque.PopFront()
+	}
+
+	return validCount, nil
```

Comment:
Pedant comment: it looks like if we get here validCount == currSize, whereas in case any error occurs we always return 0 as number of processed records. myabe we can just return currSize?

Reply:
the only case is when casting hit error somehow in the current code
```
		record, ok := ch.deque.At(i).(*ClickHouseFlowRow)
		if !ok {
			continue
		}
```
but overall i think you're right. let me change it